addons/web/static/src/legacy/js/public/public_root.js

```diff
@@ -328,6 +348,46 @@ var PublicRoot = publicWidget.RootWidget.extend({
     },
 });
 
-export default {
-    PublicRoot: PublicRoot,
-};
+const { Component, mount } = owl;
+
+/**
+ * Configure Owl with the public env
+ */
+owl.config.mode = legacyEnv.isDebug() ? "dev" : "prod";
+owl.Component.env = legacyEnv;
+
+/**
+ * This widget is important, because the tour manager needs a root widget in
+ * order to work. The root widget must be a service provider with the ajax
+ * service, so that the tour manager can let the server know when tours have
+ * been consumed.
+ */
+export async function createPublicRoot(RootWidget) {
+    await lazyloader.allScriptsLoaded;
+    AbstractService.prototype.deployServices(legacyEnv);
+    // add a bunch of mapping services that will redirect service calls from the legacy env
+    // to the wowl env
+    serviceRegistry.add("legacy_rpc", makeLegacyRpcService(legacyEnv));
+    serviceRegistry.add("legacy_session", makeLegacySessionService(legacyEnv, session));
+    serviceRegistry.add("legacy_notification", makeLegacyNotificationService(legacyEnv));
+    serviceRegistry.add("legacy_crash_manager", makeLegacyCrashManagerService(legacyEnv));
+    serviceRegistry.add("legacy_dialog_mapping", makeLegacyDialogMappingService(legacyEnv));
+    await Promise.all([owl.utils.whenReady(), session.is_bound]);
+
+    const wowlEnv = makeEnv();
+    wowlEnv.qweb.addTemplates(await loadBundleTemplates("web.assets_frontend"));
+    await startServices(wowlEnv);
+
+    const adapter = new ComponentAdapter(null, { Component }); // Used for _trigger_up compat layer
+    const publicRoot = new RootWidget(adapter);
+    await Promise.all([
+        mount(MainComponentsContainer, {
+            target: document.body,
+            env: wowlEnv,
+        }),
+        publicRoot.attachTo(document.body),
+    ]);
+    return publicRoot;
+}
```

Comment:
Ok, kinda lost in all of this but I guess it's because I don't know OWL (and unfortunately no time to check it right now). Kinda curious why the "root" container... must have a container itself? :thinking: Can't it *be* that container somehow?

Reply:
Not sure what you mean here. The publicRoot widget doesn't have a container (though it has a parent to handle trigger_up but that parent is not in the DOM).

Reply:
container/parent: my point ons that "my root stuff has a parent" = it seem strange, but ok I guess I can live with "root" "of the DOM" but not of the objects hierarchy

Reply:
Right. This is not a permanent solution, the next step will be converting the publicRoot/websiteRoot to fully fledged owl components, and in that scenario they won't need to have the ComponentAdapter as a parent anymore, they will be the actual root in both the DOM and in the component hierarchy, and then I'll work my way down to the descendants.